pyresample/test/test_geometry.py

```diff
@@ -1872,26 +1872,357 @@ def test_compute_optimal_bb(self):
         assert_np_dict_allclose(res.proj_dict, proj_dict)
         self.assertEqual(res.shape, (6, 3))
 
-    def test_aggregation(self):
+    def test_downsampling(self):
```

Comment:
These test functions are way too long and have a lot of duplicated code. Please refactor these, and try to stick with the principle that one test function should assert one thing.

Reply:
Can you elaborate a bit more? I am new to tests. 
Should I remove all testing of different array types? 
The other tests were necessary to not reduce coverage ...

Reply:
you don't need to remove anything, everything you test here is valid of course. What I mean is that it's easier to read and debug if you keep small tests. So this test function here should be split in many smaller test functions, for example `test_downsampling_keeps_ndarrays` that would just do `assert isinstance(res_np.lats, np.ndarray)`, then another function called eg `test_downsampling_keeps_DataArray` with `assert isinstance(res_xr.lats, xr.DataArray)`, etc.
All the code that will be duplicated between the smaller test should be put in a common `setup` method, so you might need to create more test cases.
I'm advising that each test should just test one behaviour of what function you are testing.

Reply:
I tried to follow your suggestions. Have a look at the `TestSwathDefinitionDownsampling` class in `test_geometry` and tell me if that is what you expect.
I wait for your feedbacks ... and then I will adapt also the rest of the tests. 
Is not clear to me if I need to create a test class for each method in the end ...